app/uk/gov/hmrc/mobilemessages/connector/MessageConnector.scala

```diff
@@ -67,7 +67,7 @@ class MessageConnector @Inject()(
     val authToken: Authorization = hc.authorization.getOrElse(throw new IllegalArgumentException("Failed to find auth header!"))
     val userId = auth.getOrElse(throw new IllegalArgumentException("Failed to find the user!"))
 
-    val keys = Seq(SessionKeys.authToken -> encode(authToken.value, UTF_8), SessionKeys.userId -> userId.authId)
+    val keys = Seq(SessionKeys.authToken -> encode(authToken.value, UTF_8), SessionKeys.userId -> userId.userID.get)
```

Comment:
Not sure you need to be doing `userId.userID.get`, it won't throw the nicest of exceptions if `None`. Also on line 68 you are already throwing an exception if the auth object is `None` ... and it doesn't look like the auth object gets used for anything else, so you could probably just get the userId out before hand and throw and exception there instead of doing `.get`